jenkins/_jenkins.py

```diff
@@ -91,7 +110,34 @@ def _flatten_builds(builds):
             yield thing
 
 
-def make_data_frame(builds):
+def make_build_data_frame(builds):
+    """
+    Make a DataFrame from of top-level build information.
+
+    :param Iterable[dict] builds: an iterable of
+        dicts of build information.
+    :return pandas.DataFrame: a DataFrame containing
+        that data augmented with week number
+        and numeric result.
+    """
+    return pandas.DataFrame(builds).assign(
+        week_number=make_week_numbers,
+        numeric_result=make_numeric_results,
+    )
+
+
+def make_subbuild_data_frame(builds):
+    """
+    Make a DataFrame from of sub build information.
+
+    This indexes all builds that are triggered by
+    the top level builds.
+
+    :param Iterable[dict] builds: an iterable of
+        dicts of build information.
```

Comment:
Yes, pretty much. It's possible the other one wouldn't have the sub-build
information.

Any ideas for how to define a type that is "dict with <these> keys?"